File: wanandroid/src/main/java/com/oreooo/wanandroid/search/SearchAct.java
package com.oreooo.wanandroid.search;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputEditText;
import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.widget.Button;
import android.widget.TextView;

import com.oreooo.baselibrary.listener.AbstractButtonClickListener;
import com.oreooo.baselibrary.list.BaseRecyclerAdapter;
import com.oreooo.baselibrary.mvp.BaseActivity;
import com.oreooo.baselibrary.util.StringUtil;
import com.oreooo.wanandroid.R;
import com.oreooo.baselibrary.pojo.Article;
import com.oreooo.baselibrary.pojo.ArticleDatas;

public class SearchAct extends BaseActivity implements SearchContract.View {

    private RecyclerView mSearchResultList;
    private TextInputEditText mSearchEditText;
    private Button mSearchButton;
    private SearchContract.Presenter mPresenter;
    private String mKeyword;
    private BaseRecyclerAdapter<ArticleDatas> mAdapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.act_search);
        init();
    }

    private void init() {
        mSearchResultList = findViewById(R.id.recycler_search);
        mSearchEditText = findViewById(R.id.edit_text_search);
        mSearchButton = findViewById(R.id.search_btn);


        mSearchButton.setOnClickListener(new AbstractButtonClickListener() {
            @Override
            protected void onSingleClick() {
                final String keyword = String.valueOf(mSearchEditText.getText());
                mKeyword = keyword;
                if (StringUtil.isNotEmpty(keyword)) {
                    mPresenter.getSearchArticle(0, keyword);
                } else {
                    // todo 弹出搜索关键字为空的提示？
                }
            }
        });

    }

    @Override
    public void showSearchResult(Article data) {
        if (mAdapter == null) {
            mAdapter = new BaseRecyclerAdapter<ArticleDatas>(SearchAct.this,
                    data.getData().getDatas(), R.layout.list_item_article, null) {
                @Override
                public void bindHolder(BaseViewHolder holder, ArticleDatas item, int position) {
                    ((TextView) holder.getView(R.id.txt_article_name))
                            .setText((Html.fromHtml("《" + item.getTitle() + "》")));
                }
            };
            mSearchResultList.setAdapter(mAdapter);
        } else {
            mAdapter.setNewData(data.getData().getDatas());
            mAdapter.notifyDataSetChanged();
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        subscribe();
    }

    @Override
    protected void onStop() {
        super.onStop();
        unsubscribe();
    }

    @Override
    public void subscribe() {
        this.mPresenter = SearchPresenter.getInstance();
        mPresenter.setView(this);
    }

    @Override
    public void unsubscribe() {
        this.mPresenter = null;
    }
}

File: module_search/src/main/java/com/oreo/module_search/SearchAct.kt
package com.oreo.module_search

import android.os.Bundle
import android.text.Html
import android.widget.TextView
import com.alibaba.android.arouter.facade.annotation.Route
import com.oreooo.baselibrary.list.BaseRecyclerAdapter
import com.oreooo.baselibrary.mvpbase.StartActivity
import com.oreooo.baselibrary.pojo.Article
import com.oreooo.baselibrary.pojo.ArticleDatas
import com.oreooo.baselibrary.route.RoutePath
import com.oreooo.baselibrary.util.StringUtil
import kotlinx.android.synthetic.main.act_search.*

@Route(path = RoutePath.SEARCH_ACTIVITY)
class SearchAct : StartActivity(), SearchContract.View {

    private var mPresenter: SearchContract.Presenter? = null

    // 页码， 上划加载更多时使用
    private var page = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.act_search)
        init()
    }

    fun init() {
        search_btn.setOnClickListener {
            val keyword = edit_text_search.text.toString()
            if (!StringUtil.isEmpty(keyword)) {
                mPresenter?.getSearchArticle(0, keyword)
            }
        }
    }

    override fun onStart() {
        super.onStart()
        subscribe()
    }

    override fun onStop() {
        super.onStop()
        unsubscribe()
    }

    override fun showArticleResult(data: Article) {
        if (recycler_search.adapter == null) {
            val adapter: BaseRecyclerAdapter<ArticleDatas> = object : BaseRecyclerAdapter<ArticleDatas>
            (this, data.data.datas, R.layout.list_item_article, null) {
                override fun bindHolder(holder: BaseViewHolder?, item: ArticleDatas?, position: Int) {
                    holder?.getView<TextView>(R.id.txt_article_name)?.text =
                            Html.fromHtml("《" + item!!.title + "》")
                }
            }
            recycler_search.adapter = adapter
        } else {
            val adapter = recycler_search.adapter as BaseRecyclerAdapter<ArticleDatas>
            adapter.setNewData(data.data.datas)
            adapter.notifyDataSetChanged()
        }
    }

    override fun subscribe() {
        mPresenter = SearchPresenter.instance
        mPresenter?.setView(this)
    }

    override fun unsubscribe() {
        mPresenter = null
    }
}